crates/relayer-cli/src/commands/evidence.rs

```diff
@@ -0,0 +1,362 @@
+        let res = monitor_misbehaviours(rt, chain);
+
+        match res {
+            Ok(()) => Output::success(()).exit(),
+            Err(e) => Output::error(e).exit(),
+        }
+    }
+}
+
+fn monitor_misbehaviours(rt: Arc<TokioRuntime>, mut chain: CosmosSdkChain) -> eyre::Result<()> {
+    let subscription = chain.subscribe()?;
+
+    // check previous blocks for equivocation that may have been missed
+    // let tm_latest_height = chain
+    //     .rpc_client
+    //     .status()
+    //     .await?
+    //     .sync_info
+    //     .latest_block_height;
+    //
+    // let latest_height = Height::new(chain.id().version(), tm_latest_height.value()).unwrap();
+    // let num_blocks = min(tm_latest_height.value(), 100);
+    // let mut height = latest_height;
+    //
+    // for _height in 0..num_blocks - 1 {
+    //     debug!("trying to check for evidence at height {height}");
+    //
+    //     equivocation_handling(&chain, height).await?;
+    //
+    //     height = height.decrement().unwrap();
+    // }
+
+    // process new block events
+    while let Ok(event_batch) = subscription.recv() {
+        match event_batch.deref() {
+            Ok(event_batch) => {
+                for event_with_height in &event_batch.events {
+                    if let IbcEvent::NewBlock(new_block) = &event_with_height.event {
+                        debug!("{new_block:?}");
+
+                        check_misbehaviour_at(rt.clone(), &chain, new_block.height)?;
+                    }
+                }
+            }
+            Err(e) => {
+                dbg!(e);
+            }
+        }
+    }
+
+    Ok(())
+}
+
+use tendermint_rpc::{Client, Paging};
+
+/// Check for misbehaviour evidence in the block at the given height.
+/// If such evidence is found, handle it by submitting it to all counterparty
+/// clients of the chain, freezing them.
+fn check_misbehaviour_at(
+    rt: Arc<TokioRuntime>,
+    chain: &CosmosSdkChain,
+    height: Height,
+) -> eyre::Result<()> {
+    let block = rt
+        .block_on(chain.rpc_client.block(TendermintHeight::from(height)))?
+        .block;
+
+    for evidence in block.evidence.into_vec() {
+        match evidence {
+            tendermint::evidence::Evidence::DuplicateVote(dv) => {
+                debug!("found duplicate vote evidence {dv:?}");
+            }
+            tendermint::evidence::Evidence::LightClientAttack(lc) => {
+                debug!("found light client attack evidence {lc:?}");
+
+                handle_light_client_attack(rt.clone(), chain, *lc)?;
+            }
+        }
+    }
+
+    Ok(())
+}
+
+fn handle_light_client_attack(
+    rt: Arc<TokioRuntime>,
+    chain: &CosmosSdkChain,
+    lc: LightClientAttackEvidence,
+) -> eyre::Result<()> {
+    // Build the two headers to submit as part of the `MsgSubmitMisbehaviour` message.
+    let (header1, header2) = build_evidence_headers(rt.clone(), chain, lc)?;
```

Comment:
we should also build a client update for the common header and then send it to those clients that don't have the consensus state associated with the common header. Otherwise the evidence cannot be verified on chain.